maat-court-data-api/src/main/java/gov/uk/courtdata/dces/service/ConcorContributionsService.java

```diff
@@ -0,0 +1,99 @@
+package gov.uk.courtdata.dces.service;
+
+import static gov.uk.courtdata.dces.enums.ConcorContributionStatus.SENT;
+
+import gov.uk.courtdata.dces.enums.ConcorContributionStatus;
+import gov.uk.courtdata.dces.mapper.ContributionFileMapper;
+import gov.uk.courtdata.dces.request.ConcorContributionRequest;
+import gov.uk.courtdata.entity.ConcorContributionsEntity;
+import gov.uk.courtdata.entity.ContributionFilesEntity;
+import gov.uk.courtdata.exception.MAATCourtDataException;
+import gov.uk.courtdata.repository.ConcorContributionsRepository;
+import gov.uk.courtdata.repository.ContributionFilesRepository;
+import gov.uk.courtdata.util.ValidationUtils;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Optional;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ConcorContributionsService {
+    private static final String DB_USER_NAME = "MLA";
+    private final ConcorContributionsRepository concorRepository;
+    private final ContributionFilesRepository contributionFileRepository;
+    private final ContributionFileMapper contributionFileMapper;
+
+    public List<String> getConcorFiles(final ConcorContributionStatus status) {
+        log.info("Getting concor contribution file with status with the -> {}", status);
+        final List<ConcorContributionsEntity> concorFileList = concorRepository.findByStatus(status);
+        return concorFileList.stream().map(ConcorContributionsEntity::getCurrentXml).toList();
+    }
+
+    @Transactional(rollbackFor = MAATCourtDataException.class)
+    public boolean createContributionAndUpdateConcorStatus(final ConcorContributionRequest contributionRequest){
+
+        ValidationUtils.isNull(contributionRequest,"contributionRequest object is null");
+        ValidationUtils.isEmpty(contributionRequest.getContributionIds(),"ContributionIds are empty/null.");
+
+        final ContributionFilesEntity contributionFilesEntity = createContributionFile(contributionRequest);
```

Comment:
It's my understanding that this method `createContributionAndUpdateConcorStatus` will **NOT** roll back both DB changes in the case where the `ContributionFilesEntity` is successfully persisted but the number of `ConcorContributionsEntity` found via ids are **zero**

Can you confirm that this is the expected behavior?